app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java

```diff
@@ -0,0 +1,427 @@
+    File noteFile = new File(Environment.getExternalStorageDirectory() + "/Kiwix/Notes/" + zimFileTitle + "/" + articleTitle + ".txt");
+
+    if(noteFile.exists()) {
+      noteFileExists = true;
+
+      StringBuilder contents = new StringBuilder();
+      try {
+
+        BufferedReader input = new BufferedReader(new java.io.FileReader(noteFile));
+        try {
+          String line = null;
+
+          while((line = input.readLine()) != null) {
+            contents.append(line);
+            contents.append(System.getProperty("line.separator"));
+          }
+        } catch (IOException e) {
+          e.printStackTrace();
+          Log.d(TAG, "Error reading line with BufferedReader");
+        } finally {
+          input.close();
+        }
+
+      } catch (IOException e) {
+        e.printStackTrace();
+        Log.d(TAG, "Error closing BufferedReader");
+      }
+
+      addNoteEditText.setText(contents.toString()); // Display the note content
+
+      enableShareNoteMenuItem(); // As note content exists which can be shared
+    }
+
+    // No action in case the note file for the currently open article doesn't exist
+  }
+
+  private void shareNote() {
+
+    /* The note text file corresponding to the currently open article, given at:
+     *    "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+     * is shared via an app-chooser intent
+     * */
+
+    if(noteEdited) {
+     saveNote(addNoteEditText.getText().toString()); // Save edited note before sharing the text file
+    }
+
+    File noteFile = new File(Environment.getExternalStorageDirectory() + "/Kiwix/Notes/" + zimFileTitle + "/" + articleTitle + ".txt");
+
+    Uri noteFileUri = null;
+    if(noteFile.exists()) {
+
+      if (Build.VERSION.SDK_INT >= 24) {
+        // From Nougat 7 (API 24) access to files is shared temporarily with other apps
+        // Need to use FileProvider for the same
+        noteFileUri = FileProvider.getUriForFile(getContext(), BuildConfig.APPLICATION_ID+".fileprovider", noteFile);
+
+      } else {
+        noteFileUri = Uri.fromFile(noteFile);
+      }
+
+    } else {
+      Toast.makeText(getContext(), getString(R.string.note_share_error_file_missing), Toast.LENGTH_SHORT).show();
+    }
+
+    if(noteFileUri != null) {
+      Intent noteFileShareIntent = new Intent(Intent.ACTION_SEND);
+      noteFileShareIntent.setType("application/octet-stream");
+      noteFileShareIntent.putExtra(Intent.EXTRA_STREAM, noteFileUri);
+      noteFileShareIntent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+      Intent shareChooser = Intent.createChooser(noteFileShareIntent, getString(R.string.note_share_app_chooser_title));
+
+      if(noteFileShareIntent.resolveActivity(getActivity().getPackageManager()) != null) {
+        startActivity(shareChooser);
+      }
+    }
+  }
+
+  public static boolean isExternalStorageWritable() {
+    String state = Environment.getExternalStorageState();
+    if(Environment.MEDIA_MOUNTED.equals(state)) {
```

Comment:
it is exactly the same to just say `return Environment.MEDIA_MOUNTED.equals(Environment.getExternalStorageState()`

Reply:
Refactored [here](https://github.com/kiwix/kiwix-android/pull/1198/commits/873eb2bce86c76b11962367701cb66bdf66b9f5e#diff-36e13fa4a9132616aee81e1f88bfa0ffR358)